discover/deployment/articles-dxp/01-installing-search-engine/06-xpack-security.markdown

```diff
@@ -0,0 +1,240 @@
+The `setup-passwords` command is the simplest method to set the built-in users'
+first-use passwords for the first time. To update a password subsequently, use
+Kibana's UI or the 
+[Change Password API](https://www.elastic.co/guide/en/elasticsearch/reference/6.1/security-api-change-password.html).
+
+The `interactive` argument lets you set the passwords for all built-in users.
+The configuraiton shown in these articles assumes you set all of the
+passwords to *liferay*. Of course, that's not recommended for production systems.
+
+    ./bin/x-pack/setup-passwords interactive
+
+See Elastic's documentation on the 
+[setup-passwords command](https://www.elastic.co/guide/en/elasticsearch/reference/6.1/setup-passwords.html) 
+for additional options.
+
+Since you're securing Elasticsearch, make sure you keep track of the password
+set for the `elastic` user. 
+
+We recommend enabling transport layer security on each node.
+
+## Enabling Transport Layer Security [](id=enabling-transport-layer-security)
+
+The following instructions for enabling TLS use `liferay` as the password
+whenever one is needed. Customize these as appropriate for your installation. 
+
+### Generate Node Certificates [](id=generate-node-certificates)
+
+[Generate a node certificate](https://www.elastic.co/guide/en/elasticsearch/reference/6.1/configuring-tls.html#node-certificates)
+for each node. You can, of course, use a Certificate Authority toobtain node 
+certificates.
+
+1.  Create a certificate authority, using 
+      [X-Pack's `certutil`](https://www.elastic.co/guide/en/elasticsearch/reference/6.1/certutil.html)
+      command:
+
+        ./bin/x-pack/certutil ca --pem --ca-dn CN=localhost
+
+    This generates a ZIP file. Unzip the contents somewhere safe.
+
+2. Generate X.509 certificates and private keys using the CA from Step 1. For
+   example:
+
+        ./bin/x-pack/certutil cert --pem --ca-cert /path/to/ca.crt --ca-key /path/to/ca.key --dns localhost --ip 127.0.0.1 --name localhost
+
+    This generates another ZIP file. Extract the contents somewhere in the
+    `Elasticsearch Home/config` folder.
+
++$$$
+
+**Note:** The `certutil` command defaults to using the *PKSC#12* format
+for certificate generation. Kibana does not work with PKSC#12
+certificates, so the `--pem` option (to generate the certificate in PEM
+format) is important if you're using X-Pack monitoring.
+
+$$$
+
+### Enable TLS [](id=enable-tls)
+
+[Enable TLS](https://www.elastic.co/guide/en/elasticsearch/reference/6.1/configuring-tls.html#enable-ssl) 
+on each node via its `elasticsearch.yml`.
+
+1.  Add the certificate, key and certificate authority paths to each node's
+    `elasticsearch.yml`:
+
+        xpack.ssl.certificate: /path/to/[Elasticsearch Home]/config/localhost.crt
+        xpack.ssl.key: /path/to/[Elasticsearch Home]/config/localhost.key
+        xpack.ssl.certificate_authorities: ["/path/to/ca.crt"]
```

Comment:
On restart of Elasticsearch after installing X-Pack got log exceptions caused by:

`Caused by: java.security.AccessControlException: access denied ("java.io.FilePermission" "/home/russell/liferay-bundles/master-es6-xpack/elasticsearch-6.1.3/configca/ca.crt" "read")`

Reply:
Must be under config dir, should change example path to show this, at a minimum

Reply:
No, I think this was likely just due to the bad path I entered:
configca/ca.crt should have been config/ca/ca.crt
disregarding